src/Common/src/Internal/Runtime/TypeLoader/EETypeOptionalFields.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+using Internal.Runtime;
+using Internal.NativeFormat;
+using System.Diagnostics;
+using System.Text;
+
+namespace Internal.Runtime.TypeLoader
+{
+    enum OptionalFieldTag : byte
+    {
+        OFT_RareFlags,
+        OFT_ICastableIsInstSlot,
+        OFT_DispatchMap,
+        OFT_ValueTypeFieldPadding,
+        OFT_ICastableGetImplTypeSlot,
+        OFT_NullableValueOffset,
+
+        OFT_Count // Number of field types we support
+    }
```

Comment:
Just a thought - maybe it would be better to have all these EEType-related constants in a single file.